app.rb

```diff
@@ -1,8 +1,35 @@
 require 'sinatra/base'
+
+
 class RockPaperScissors < Sinatra::Base
-  get '/test' do
-    'test page'
+
+  enable :sessions
+
+  # get '/test' do
+  #   'home page'
+  # end
+
+  get '/' do
+   erb :index
+  end
+
+	post '/names' do
+		session[:player_name] = params[:player_name]
+  	redirect '/play'
+	end
+
+  get '/play' do
+    @player_name = session[:player_name] 
+    erb :play
   end
 
+	get '/result' do
+		@player_name = session[:player_name] 
+		erb :result 
```

Comment:
Before this, you should access the button/input value from the user (rock, paper or scissors).

You could then sample the three choices for the computer, and work out a result before returning to the Result page.